### PR TITLE
Set up Isomalloc mmap backend on Windows

### PR DESCRIPTION
**Pull request by evan-charmworks**

### CODE REVIEW

File: src/conv-core/isomalloc.C [hunk 1]
```diff
@@ -168,6 +182,8 @@ static int init_map() { return 0; /*Isomalloc never works without mmap*/ }
 CpvStaticDeclare(int, zerofd); /*File descriptor for /dev/zero, for mmap*/
 #endif
 
+static constexpr void * const mmap_fail = (void *)~(uintptr_t)0;
```
Comment (matthiasdiener) [inline src/conv-core/isomalloc.C]: the value of mmap_fail is different here from the value for _Win32. Is this intentional?

Comment (evan-charmworks) [reply src/conv-core/isomalloc.C]: Yes, it varies by implementation. mmap returns -1, VirtualAlloc returns null.

Comment (matthiasdiener) [reply src/conv-core/isomalloc.C]: Ok, can you add a brief comment there to explain this? Also, could this be `MAP_FAILED` instead of -1?

Comment (evan-charmworks) [reply src/conv-core/isomalloc.C]: Yes to both, done

**Pull request merged**